frontend/src/stimulus/controllers/dynamic/work-packages/activities-tab/internal-comment.controller.ts

```
@@ -149,7 +153,44 @@ export default class InternalCommentController extends BaseController {
     });
   }
 
+  private persistInternalState(isChecked:boolean):void {
+    try {
+      localStorage.setItem(this.storageKey, JSON.stringify(isChecked));
+    } catch (error) {
+      console.warn('Failed to persist internal comment state:', error);
```

Do we have any recommendation or pattern on how to handle errors on frontend code like this? I feel this warning and the one on `restoreInternalState` really don't serve much of a purpose.

---

Hmm, I’m not sure if there’s an established pattern. I think error logging makes things visible and mitigates silent failures. 🤔

---

AFAICS, we do have a wrapper around `console.log` [`debugLog`](https://github.com/opf/openproject/blob/c7b634e5edc5fc673ac6a68991384faddc492e1f/frontend/src/app/shared/helpers/debug_output.ts) that might be more conventional. As it only outputs errors when debugging.